source/Engine.cpp

```diff
@@ -928,6 +932,25 @@ void Engine::Draw() const
 	draw[drawTickTock].Draw();
 	batchDraw[drawTickTock].Draw();
 
+	Point dCenter = player.FlagshipPtr()->Position();
+	for(const shared_ptr<Ship> &ship : ships)
+	{
+		if(ship->Cloaking() && ship->IsYours() && ship->GetSystem() == player.GetSystem())
+		{
+			float cloak = .1f * static_cast<float>(ship->Cloaking());
+			OutlineShader::Draw(ship->GetSprite(), (ship->Position() - dCenter) * zoom, Point(ship->Width(), ship->Height()) * zoom,
+								Color(6 * cloak, cloak, cloak, 8 * cloak), ship->Facing().Unit(), ship->GetFrame());
+		}
+
+		if(Preferences::Has("Damage highlights"))
```

Comment:
How about we store this in a bool before the loop instead of checking for each ship?